compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTest.scala

```diff
@@ -0,0 +1,70 @@
+package dotty.tools
+package dotc
+package reporting
+
+import diagnostic._
+import core.Contexts.Context
+
+import scala.collection.mutable
+
+import org.junit.Assert._
+
+trait ErrorMessagesTest extends DottyTest {
+
+  class Report(messages: List[Message], ictx: Context) {
+    def expect(f: (Context, List[Message]) => Unit): Unit = {
+      f(ictx, messages)
+    }
+
+    def expectNoErrors: Unit = this.isInstanceOf[FailedReport]
```

Comment:
Then I don't understand the statement, this computes a `Boolean` and converts it to `Unit` regardless of it's value?